Tests/DatadogTests/Datadog/CrashReporting/CrashContext/CrashContextProviderTests.swift

```diff
@@ -14,355 +14,128 @@ import CoreTelephony
 /// separate part of the `CrashContext` information.
 class CrashContextProviderTests: XCTestCase {
-    // MARK: - `TrackingConsent` Integration
+    // MARK: - `DatadogContext` Integration
 
     func testWhenTrackingConsentValueChangesInConsentProvider_thenCrashContextProviderNotifiesNewContext() {
         let expectation = self.expectation(description: "Notify new crash context")
-        let initialTrackingConsent: TrackingConsent = .mockRandom()
-        let randomTrackingConsent: TrackingConsent = .mockRandom(otherThan: initialTrackingConsent)
 
-        let trackingConsentProvider = ConsentProvider(initialConsent: initialTrackingConsent)
-        let crashContextProvider = CrashContextProvider(
-            consentProvider: trackingConsentProvider,
-            userInfoProvider: .mockAny(),
-            networkConnectionInfoProvider: NetworkConnectionInfoProviderMock.mockAny(),
-            carrierInfoProvider: CarrierInfoProviderMock.mockAny(),
-            rumViewEventProvider: .mockRandom(),
-            rumSessionStateProvider: .mockAny(),
-            appStateListener: AppStateListenerMock.mockAny()
-        )
-
-        let initialContext = crashContextProvider.currentCrashContext
-        var updatedContext: CrashContext?
+        // Given
+        let crashContextProvider = CrashContextProvider()
+        let core = PassthroughCoreMock(messageReceiver: crashContextProvider)
+        let context: DatadogContext = .mockRandom()
 
         // When
-        crashContextProvider.onCrashContextChange = { newContext in
-            updatedContext = newContext
+        crashContextProvider.onCrashContextChange = {
+            XCTAssertEqual($0.serverTimeOffset, context.serverTimeOffset)
+            XCTAssertEqual($0.service, context.service)
+            XCTAssertEqual($0.env, context.env)
+            XCTAssertEqual($0.version, context.version)
+            XCTAssertEqual($0.device.osVersion, context.device.osVersion)
+            XCTAssertEqual($0.sdkVersion, context.sdkVersion)
+            XCTAssertEqual($0.source, context.source)
+            XCTAssertEqual($0.trackingConsent, context.trackingConsent)
+            XCTAssertEqual($0.userInfo, context.userInfo)
+            XCTAssertEqual($0.networkConnectionInfo, context.networkConnectionInfo)
+            XCTAssertEqual($0.carrierInfo, context.carrierInfo)
+            XCTAssertEqual($0.lastIsAppInForeground, context.applicationStateHistory.currentSnapshot.state.isRunningInForeground)
             expectation.fulfill()
         }
-        trackingConsentProvider.changeConsent(to: randomTrackingConsent)
+
+        core.send(message: .context(context))
 
         // Then
-        waitForExpectations(timeout: 1, handler: nil)
-        XCTAssertEqual(initialContext.lastTrackingConsent, initialTrackingConsent)
-        XCTAssertEqual(updatedContext?.lastTrackingConsent, randomTrackingConsent)
+        waitForExpectations(timeout: 0.5, handler: nil)
     }
```

Comment:
The name of this test does no longer reflect what it does. I think it should be rather saying: _**when** SDK context changes, **then** then it notifies new crash context_.